tests/core/test_i64.in

```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
```

Comment:
My only issue with having the extension be `.in` is that it won't be obvious whether it's C or C++ which sometimes matters. Maybe we could make it `test_foo.c` and `test_foo.c.out` where it's the presence of the `out` file that actually defines what tests exist?